connect/runtime/src/test/java/org/apache/kafka/connect/integration/ConnectWorkerIntegrationTest.java

```diff
@@ -180,6 +187,56 @@ public void testRestartFailedTask() throws Exception {
             CONNECTOR_SETUP_DURATION_MS, "Connector tasks are not all in running state.");
     }
 
+    /**
+     * Verify that a set of tasks restarts correctly after a broker goes offline and back online
+     */
+    @Test
+    public void testBrokerCoordinator() throws Exception {
+        workerProps.put(DistributedConfig.SCHEDULED_REBALANCE_MAX_DELAY_MS_CONFIG, String.valueOf(5000));
+        connect = connectBuilder.workerProps(workerProps).build();
+        // start the clusters
+        connect.start();
+        int numTasks = 4;
+        // create test topic
+        connect.kafka().createTopic("test-topic", NUM_TOPIC_PARTITIONS);
+
+        // setup up props for the sink connector
+        Map<String, String> props = new HashMap<>();
+        props.put(CONNECTOR_CLASS_CONFIG, MonitorableSourceConnector.class.getSimpleName());
+        props.put(TASKS_MAX_CONFIG, String.valueOf(numTasks));
+        props.put("topic", "test-topic");
+        props.put("throughput", String.valueOf(1));
+        props.put("messages.per.poll", String.valueOf(10));
+        props.put(KEY_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+
+        waitForCondition(() -> assertWorkersUp(NUM_WORKERS).orElse(false),
+                WORKER_SETUP_DURATION_MS, "Initial group of workers did not start in time.");
+
+        // start a source connector
+        connect.configureConnector(CONNECTOR_NAME, props);
+
+        waitForCondition(() -> assertConnectorAndTasksRunning(CONNECTOR_NAME, numTasks).orElse(false),
+                CONNECTOR_SETUP_DURATION_MS, "Connector tasks did not start in time.");
+
+        connect.kafka().stopOnlyKafka();
+
+        waitForCondition(() -> assertWorkersUp(NUM_WORKERS).orElse(false),
+                WORKER_SETUP_DURATION_MS, "Group of workers did not remain the same after broker shutdown");
+
+        connect.kafka().startOnlyKafkaOnSamePorts();
+
+        // Allow for the workers to discover that the coordinator is unavailable
+        Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```

Comment:
Shouldn't the workers discover that the coordinator is unavailable while it is down?
I'm imagining this test going like this:

1. steady-state workers are running
2. brokers stop
3. workers discover the coordinator is unavailable 
4. workers stop their tasks
5. brokers start
6. workers discover the next coordinator 
7. workers start their tasks
8. workers are running unaffected

Reply:
That was a bit misplaced, because I actually need 3 explicit delays (due to current lack of appropriate handles from the kafka and connect embedded clusters). 
1. Bring kafka down, allow workers to discover it's down (heartbeat * 2 + 4 sec)
2. Allow for Kafka to come back up
3. Allow for worker cluster to stabilize after the very last rebalance (delay = 5sec) 

Added another commit.